Stratis.Bitcoin/FullNode.cs

```diff
@@ -20,23 +20,27 @@
 using Stratis.Bitcoin.Miner;
 using Stratis.Bitcoin.Utilities;
 using System.Reflection;
+using System.Runtime.Loader;
 
 namespace Stratis.Bitcoin
 {
 
-	public class FullNode : IFullNode, IDisposable
+	public class FullNode : IFullNode
 	{
-		private ApplicationLifetime applicationLifetime; // this will replace the cancellation token on the full node
-		ILogger _logger;
-		private FullNodeFeatureExecutor fullNodeFeatureExecutor;
+		private ApplicationLifetime _applicationLifetime; // this will replace the cancellation token on the full node
+		private readonly ILogger _logger;
+		private FullNodeFeatureExecutor _fullNodeFeatureExecutor;
+	    private bool _stopped;
+	    private readonly ManualResetEvent _isDisposed = new ManualResetEvent(false);
+	    private readonly ManualResetEvent _isStarted = new ManualResetEvent(false);
 
-		public IFullNodeServiceProvider Services { get; set; }
+        public IFullNodeServiceProvider Services { get; set; }
 
-		NodeSettings _Settings;
+		NodeSettings _settings;
```

Comment:
If you are already changing the full node class maybe remove all the "_" in the parameters